library/Backends/llvm/src/Juvix/Backends/LLVM/Compilation.hs

```diff
@@ -1,65 +1,216 @@
+{-# LANGUAGE NamedFieldPuns #-}
 {-# LANGUAGE OverloadedStrings #-}
 
 module Juvix.Backends.LLVM.Compilation
   ( compileProgram,
   )
 where
 
+import qualified Data.String as S (fromString)
 import Juvix.Backends.LLVM.Primitive
 import qualified Juvix.Core.Erased.Ann as ErasedAnn
 import Juvix.Library
 import Juvix.Library.Feedback
-import qualified LLVM.AST as LLVM (Module, Operand)
+import qualified Juvix.Library.HashMap as Map
+import qualified Juvix.Library.NameSymbol as NameSymbol
+import qualified LLVM.AST as LLVM (Module, Name, Operand (..))
+import qualified LLVM.AST.Constant as LLVM (Constant (..))
 import qualified LLVM.AST.Type as LLVM
 import qualified LLVM.IRBuilder as LLVM
 import qualified LLVM.Pretty as LLVM
 import qualified Prelude as P
 
+-- | A mapping between Juvix variable names and their operands in LLVM.
+type Env = Map.Map NameSymbol.T LLVM.Operand
+
+-- | Compile the input program to an LLVM module.
+-- TODO: maybe do something smarter with the module name?
```

Comment:
at some point we might want to map juvix modules to llvm modules or at least include metadata

this is low-priority however

Reply:
Yeah, currently there is no way to do so either. Everything is currently a single module in the input program, and there is no way to access a module or input filename.